ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -95,6 +95,11 @@ public void init() {
         super.init();
 
         setAttribute(AMBARI_SERVER, addChild(createServerSpec(getConfig(SECURITY_GROUP))));
+        if(!getConfig(SERVER_COMPONENTS).isEmpty()) {
+            getAttribute(AMBARI_SERVER).config().set(SoftwareProcess.CHILDREN_STARTABLE_MODE, SoftwareProcess.ChildStartableMode.BACKGROUND_LATE);
+            EntitySpec<? extends AmbariAgent> agentSpec = AmbariAgentImpl.createAgentSpec(this, null);
+            getAttribute(AMBARI_SERVER).addChild(agentSpec);
+        }
```

Comment:
Instead of using the `getAttribute(AMBARI_SERVER)`, I would loop over `getAmbariServers()` to make that all servers have the same configuration.